docs/model-dev-guide/apis-howto/api-core-ug.rst

```diff
@@ -1,214 +1,299 @@
+   be different. For example, for instructions on how to run a remote distributed training job,
+   visit the :ref:`qs-mdldev`.
 
-#. Begin by importing import the ``determined`` module:
+In the WebUI, select your experiment. You'll notice the tabs do not yet contain any information. In
+the next section, we'll report training and validation metrics.
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.py
-      :language: python
-      :start-after: NEW: import determined
-      :end-before: def main
+************************
+ Step 2: Report Metrics
+************************
 
-#. Enable ``logging``, using the ``det.LOG_FORMAT`` for logs. This enables useful log messages from
-   the ``determined`` library, and ``det.LOG_FORMAT`` enables filter-by-level in the WebUI.
+To report training and validation metrics to the Determined master, we’ll add a few lines of code to
+our script. More specifically, we'll create a :class:`~determined.core.Context` object to allow
+interaction with the master. Then, we'll pass the ``core_context`` as an argument into ``main()``,
+``train()``, and ``test()`` and modify the function headers accordingly.
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.py
-      :language: python
-      :start-at: logging.basicConfig
-      :end-at: logging.error
+To run our experiment, we'll use the ``model_def_metrics.py`` script and its accompanying
+``metrics.yaml`` experiment configuration file.
 
-#. In your ``if __name__ == "__main__"`` block, wrap the entire execution of ``main()`` within the
-   scope of :meth:`determined.core.init`, which prepares resources for training and cleans them up
-   afterward. Add the ``core_context`` as a new argument to ``main()`` because the Core API is
-   accessed through the ``core_context`` object.
+.. include:: ../../_shared/note-premade-tutorial-script.txt
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.py
-      :language: python
-      :start-at: with det.core.init
+Begin by importing Determined:
 
-#. Within ``main()``, add two calls: (1) report training metrics periodically during training and
-   (2) report validation metrics every time a validation runs.
+.. code:: python
 
-   .. literalinclude:: ../../../examples/tutorials/core_api/1_metrics.py
-      :language: python
-      :pyobject: main
+   import determined as det
 
-   The ``report_validation_metrics()`` call typically happens after the validation step, however,
-   actual validation is not demonstrated by this example.
+Step 2.1: Modify the Main Loop
+==============================
 
-#. Create a ``1_metrics.yaml`` file with an ``entrypoint`` invoking the new ``1_metrics.py`` file.
-   You can copy the ``0_start.yaml`` configuration file and change the first couple of lines:
+We'll need a :class:`~determined.core.Context` object for interacting with the master. To accomplish
+this, we'll modify the __main__loop to include ``core_context``:
```

Comment:
nit: no space between "__main__" and "loop" , and might want to put `__main__` in code style

Reply:
resolved